Util/RandomNumberGeneratorExtension.cs

```diff
@@ -54,4 +54,9 @@ public static T PickRandomUnchecked<T>(this RandomNumberGenerator rng, IEnumerab
 			.Skip(rng.RandiRange(0, values.Count() - 1))
 			.First();
 	}
+
+	public static float RandomWithVariation(this RandomNumberGenerator rng, float baseChance, float variationPercent) {
+		var v = rng.RandfRange(-variationPercent, variationPercent);
+		return baseChance * (1.0f + v);
+	}
```

Comment:
I find the naming here is very misleading. The name suggests that this is a method for obtaining a randomized number with some random variation, from which I expected something like `randf(baseChance * randf(-variation, variation))`, not `value * (1 + randf(-variation, variation))`.

As the function seems to not be about obtaining a randomized number with a variation/bias applied, but rather applying a randomized variation to a value, I think e.g. `ApplyRandomVariation` could be more descriptive.

One source of confusion is at least that in the current formula, the `baseChance` is just a base value that gets varied and (when contained within the scope of the function) has nothing to do with chances. I'd suggest renaming it to e.g. just "value".

```suggestion
	public static float ApplyRandomVariation(this RandomNumberGenerator rng, float value, float variationPercent) {
		var variation = rng.RandfRange(-variationPercent, variationPercent);
		return value * (1.0f + variation);
	}
```